drivers/adc/adc_stm32.c

```diff
@@ -413,6 +413,68 @@ static int start_read(const struct device *dev,
 	LL_ADC_SetResolution(adc, resolution);
 #endif
 
+#if defined(CONFIG_SOC_SERIES_STM32G0X) || \
+	defined(CONFIG_SOC_SERIES_STM32G4X) || \
+	defined(CONFIG_SOC_SERIES_STM32H7X) || \
+	defined(CONFIG_SOC_SERIES_STM32L0X) || \
+	defined(CONFIG_SOC_SERIES_STM32L4X) || \
+	defined(CONFIG_SOC_SERIES_STM32WBX) || \
+	defined(CONFIG_SOC_SERIES_STM32WLX)
+	switch (sequence->oversampling) {
+	case 0:
+		LL_ADC_SetOverSamplingScope(adc, LL_ADC_OVS_DISABLE);
+		break;
+	case 1:
+		LL_ADC_SetOverSamplingScope(adc, LL_ADC_OVS_GRP_REGULAR_CONTINUED);
+		LL_ADC_ConfigOverSamplingRatioShift(adc, LL_ADC_OVS_RATIO_2,
+						    LL_ADC_OVS_SHIFT_RIGHT_1);
```

Comment:
Have you tried to consider something like follows:
```
		LL_ADC_ConfigOverSamplingRatioShift(adc, (1 << sequence->oversampling + y),
						    (1 << sequence->oversampling +x));
```
to get rid of the switch case ?

Reply:
Dismissing this point, as this is minor vs the benefit of the change